test/export/StructureDefinitionExporter.test.ts

```diff
@@ -2294,6 +2410,620 @@ describe('StructureDefinitionExporter', () => {
+      // * component[Lab].code from http://example.com/MediocreObservationCodes (extensible)
+      // * component.code from http://example.com/ImportantObservationCodes (required)
+      const labValueSet = new ValueSetRule('component[Lab].code');
+      labValueSet.valueSet = 'http://example.com/MediocreObservationCodes';
+      labValueSet.strength = 'extensible';
+      const rootValueSet = new ValueSetRule('component.code');
+      rootValueSet.valueSet = 'http://example.com/ImportantObservationCodes';
+      rootValueSet.strength = 'required';
+
+      observationWithSlice.rules.push(labValueSet, rootValueSet);
+      doc.profiles.set(observationWithSlice.name, observationWithSlice);
+      exporter.export();
+      const sd = pkg.profiles[0];
+      const rootCode = sd.findElement('Observation.component.code');
+      const labCode = sd.findElement('Observation.component:Lab.code');
+      const importantBinding = {
+        valueSet: 'http://example.com/ImportantObservationCodes',
+        strength: 'required'
+      };
+      const mediocreBinding = {
+        valueSet: 'http://example.com/MediocreObservationCodes',
+        strength: 'extensible'
+      };
+      expect(rootCode.binding).toEqual(importantBinding);
+      expect(labCode.binding).toEqual(mediocreBinding);
+    });
+
+    it('should apply ValueSetRules on the child of a sliced element, then the child of a slice, with different value sets', () => {
+      // * component[Lab].code MS // force creation of element
+      // * component.code from http://example.com/ImportantObservationCodes (required)
+      // * component[Lab].code from http://example.com/MediocreObservationCodes (extensible)
+      const flagRule = new FlagRule('component[Lab].code');
+      flagRule.mustSupport = true;
+      const rootValueSet = new ValueSetRule('component.code');
+      rootValueSet.valueSet = 'http://example.com/ImportantObservationCodes';
+      rootValueSet.strength = 'required';
+      const labValueSet = new ValueSetRule('component[Lab].code');
+      labValueSet.valueSet = 'http://example.com/MediocreObservationCodes';
+      labValueSet.strength = 'extensible';
+
+      observationWithSlice.rules.push(flagRule, rootValueSet, labValueSet);
+      doc.profiles.set(observationWithSlice.name, observationWithSlice);
+      exporter.export();
+      const sd = pkg.profiles[0];
+      const rootCode = sd.findElement('Observation.component.code');
+      const labCode = sd.findElement('Observation.component:Lab.code');
+      const importantBinding = {
+        valueSet: 'http://example.com/ImportantObservationCodes',
+        strength: 'required'
+      };
+      const mediocreBinding = {
+        valueSet: 'http://example.com/MediocreObservationCodes',
+        strength: 'extensible'
+      };
+      expect(rootCode.binding).toEqual(importantBinding);
+      expect(labCode.binding).toEqual(mediocreBinding);
+    });
```

Comment:
On the non-child slices, it looks like we log an error when binding the same value set to a sliced element and the root but binding it more weakly on the sliced element (the test starting on line 2675). Should there be an error logging for a similar situation for children of slices? I'm thinking of something like:
```
* component.code from http://example.com/RegularObservationCodes (required)
* component[Lab].code from http://example.com/RegularObservationCodes (extensible)
```
There's a very good chance I'm missing something and you already tested this somewhere or this should be allowed.

Reply:
Nope, I don't have this test yet! I'll add it in.